### PR TITLE
include/posix: incorporate toolchain-provided time.h header

### LINKED ISSUE #24730: C standard library <time.h> functions and structures not available when using POSIX API
**Describe the bug**
C standard library time.h functions and structures are not available when using CONFIG_POSIX_API=y

**To Reproduce**
1. cd samples/posix/gettimeofday
2. modify src/main:
```
#include <time.h>

// ...

void main(void)
{
    time_t now = time(NULL);
    struct tm tm_now;
    localtime_r(&now, &tm_now);
    printf("Time: %s\n", asctime(&tm_now));
    // ...
}
```
3. west build -b mimxrt1064_evk
4. See error and warnings (struct tm is of unknown size and implicit declarations)

**Expected behavior**
Functions and structures are correctly defined and no error or warning shows up.

**Impact**
I am trying to port a library to Zephyr and I need to use both posix time functions and C time functions.

**Screenshots or console output**
```
../src/main.c: In function 'main':
../src/main.c:18:15: warning: implicit declaration of function 'time' [-Wimplicit-function-declaration]
   18 |  time_t now = time(NULL);
      |               ^~~~
../src/main.c:19:12: error: storage size of 'tm_now' isn't known
   19 |  struct tm tm_now;
      |            ^~~~~~
../src/main.c:20:2: warning: implicit declaration of function 'localtime_r' [-Wimplicit-function-declaration]
   20 |  localtime_r(&now, &tm_now);
      |  ^~~~~~~~~~~
../src/main.c:22:23: warning: implicit declaration of function 'asctime' [-Wimplicit-function-declaration]
   22 |  printf("time: %s\n", asctime(&tm_now));
      |                       ^~~~~~~
../src/main.c:19:12: warning: unused variable 'tm_now' [-Wunused-variable]
   19 |  struct tm tm_now;
      |            ^~~~~~
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
OS: Linux Ubuntu 18.04
Toolchain: Zephyr SDK version 0.11.2
tag: zephyr-v2.2.0

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by pabigot**
Some of what's supposed to be in <time.h> was lost because this header attempts to define everything using more primitive include files. Instead incorporate the contents from the toolchain-provided header. Any gaps should be picked up by the legacy content present in this file, which should not conflict with the toolchain header.

An updated test confirms the problematic API is now available to Zephyr.

Closes #24730.

(This solution, which addresses a blocking problem for a Nordic-based project, has been discussed in https://github.com/zephyrproject-rtos/zephyr/issues/24730#issuecomment-620653206.  The potential follow-up cleanup is not being addressed here.)

### CODE REVIEW

**Comment by jhirsi** [comment]
Thanks @pabigot! #24730 is a blocker for me currently when porting a posix based app on top of zephyr posix api. But not with this, confirming that this would solve compilation issues that I faced and would unblock me to concentrate to the actual beef.
Hopefully this could be merged!

File: include/posix/time.h [hunk 1]
```diff
@@ -101,4 +106,9 @@ int nanosleep(const struct timespec *rqtp, struct timespec *rmtp);
 #include <syscalls/time.h>
 #endif /* CONFIG_ARCH_POSIX */
 
+#else /* ZEPHYR_INCLUDE_POSIX_TIME_H_ */
+/* Read the standard header when <posix/time.h> finds itself on the
```
Comment (pfalcon) [inline include/posix/time.h:110]: What "standard header"? If you mean "toolchain-provided header", as the commit message says, please say it so here too.

File: samples/posix/gettimeofday/src/main.c [hunk 2]
```diff
@@ -16,15 +16,19 @@ int main(void)
 
 	while (1) {
 		int res = gettimeofday(&tv, NULL);
+		time_t now = time(NULL);
```
Comment (pfalcon) [inline samples/posix/gettimeofday/src/main.c:19]: So, the sample is called "samples/posix/gettimeofday", and initially was intended to showcase gettimeofday (and not less than that, system time auto-init via SNTP). If you add more function calls there, please at least add comment like "Also exercise a few other functions at the same time."

**Comment by pfalcon** [comment]
I indeed wasn't able to look into that issue myself - started with first issues in backlog for POSIX subsys (syscalls for fd primitives), and never got out of that swamp.

My only question (beyond a couple of code comment requests) is - how this was tested? (Any more or less big 3rd-party software build with it or whatever.)

File: samples/posix/gettimeofday/src/main.c (hunk 2 above)
Comment (pabigot) [reply samples/posix/gettimeofday/src/main.c:19]: Updated the example description to note that localtime is printed as well.

**Comment by pabigot** [comment]
Review comments addressed.  It's been compile-tested via sanitycheck, and externally tested by @jhirsi.

**Review by pfalcon — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**